data/mil2525d/utilities/military-features-utilities/template-gdb-toolbox/scripts/CreateTemplateGDB/CreateTemplateGDB.py

```diff
@@ -0,0 +1,197 @@
+            
+            # Now create the Versions table in the gdb
+            
+            tableName = "Versions"
+            
+            arcpy.CreateTable_management(geodatabase, tableName)
+            
+            table = os.path.join(geodatabase, tableName)
+            arcpy.AddField_management(table, "Item", "TEXT", field_length=50)
+            arcpy.AddField_management(table, "Version", "TEXT", field_length=50)
+            
+            arcpy.CopyRows_management(outFile, table)
+            
+            os.remove(outFile)
+    except Exception as err: 
+        arcpy.AddError(traceback.format_exception_only(type(err), err)[0].rstrip())
+        
+    finally:
+        arcpy.AddMessage("Success! - Completed: CreateVersionsTable")
+            
+def createTemplateGDB(schemasFolder, destinationFolder, version):
+    
+    # Creates the specified file geodatabase of the specified version
+    
+    try:
+        arcpy.AddMessage("Starting: CreateTemplateGDB")
+        
+        # Create a new file gdb in supplied folder, using supplied schema specification  
+               
+        # Schema details are read from CSV files created by JMSML
+        
+        # Open the schemas CSV file which contains the basic structure of the gdb
+        # we are going to create
+        
+        schemaFile = os.path.join(schemasFolder, "Schemas.csv")
+        
+        with open(schemaFile, 'r') as csvFile:
+            reader = csv.reader(csvFile, dialect='excel')
+            
+            # Skip the headers
+            header = next(reader)
+            
+            # Read the second line, the line of data should be a geodatabase schema line
+            
+            line = next(reader)
+            
+            if line[0] == "SchemaContainer":
+                arcpy.AddMessage("Creating file geodatabase " + line[1] + "...")
+                
+                gdbName = line[1] + ".gdb"
+                gdbPath = os.path.join(destinationFolder, gdbName)
+                
+                arcpy.env.workspace = destinationFolder
+  
+                if(arcpy.Exists(gdbName)):
+                    arcpy.Delete_management(gdbName)
+           
+                arcpy.CreateFileGDB_management(destinationFolder, gdbName, version)
+                
+                # Create a Versions table from the contents of a CSV file
+                
+                createVersionsTable(schemasFolder, gdbPath)
+                        
+                # Read the next line of data.  It should be a dataset and its metadata
+                
+                line = next(reader)
+                
+                if line[0] == "SchemaSet":
+                    arcpy.AddMessage("Creating feature dataset " + line[1] + "...")
+                    
+                    # Fetch the dataset's spatial reference
+                    
+                    dataSet = line[1]
+                    sr = arcpy.SpatialReference(int(line[17]))
```

Comment:
If that table really has 17+ entries we will probably want to revisit that csv/table design at some point to pull out the additional metadata that is not getting used here.